### ISSUE TITLE
What about a more human-friendly format?

### ISSUE THREAD

**Issue by pombredanne** [spdx/license-list-XML#395]
I wonder if a simpler format would be better? Something like YAML for metadata and text and a simple `{{mustache-style}}` curly braces markup when needed? Eventually eschewing any markup of the format?

**Comment by bradleeedmondson** [comment]
I can't speak to YAML in particular, but I believe @myndzi, @goneall, and I had talked about formatting (presentation-layer) markup, with me asking whether it's needed at all. @myndzi's response was that it was, at least if we wanted to be able to present the license text in a way that remotely resembles the original license text.

Additionally, there are lots of XML tools that we can use to make the XML-format files more susceptible to human editing. I don't know that YAML offers as much in the way of strict validation. Perhaps you'd like to propose YAML as a supported output format, rather than the authoritative format for the master files?

**Comment by pombredanne** [comment]
@bradleeedmondson supporting multiple formats would not make sense to me. 

On thing that I cannot understand is why adding markup to plain text license texts would be needed. Wrapping a the text in a `<pre>` if and when you want to display HTML should be all that is needed, or am I missing something? Especially when these licenses are available as plain text.  This is likely demanding low value, high effort work that I have difficulties to find a good purpose for. Especially since the current markup is more or less but not exactly like HTML meaning it is not even usable as-is without a transformation.

**Comment by bradleeedmondson** [comment]
My understanding is that the XML is really intended for the "master" files, and those will be used to generate the supported formats (plaintext, RDF, JSON, etc.). You could probably add YAML to that list without much effort, once the XML format is stable. But I understood part of your initial comment to suggest changing the master files to YAML, which is something we'd want to get the entire tech team and legal team to weigh in on (and in a sense it may be a little late, so we'd want to look at the cost/benefit of transitioning). Once the XML is stable, the XML license list is complete, the tools to work with the XML are available, and the supported file formats are being generated, I don't think we'll miss the accessibility of YAML.

**Comment by goneall** [comment]
I've also thought of the master XML as more of a format internal to the legal team - not something that would be directly consumed by tools outside of the SPDX community.  There is a tool that translates the master file format into 5 different consumable formats including JSON and HTML.  If there is interest, we could add YAML with a modest amount of effort.

@pombredanne to answer your earlier question, we did spend some time discussing different format before going down this path.  No one was a big fan of XML, however, the additional structure provided by XML to tag items like titles, copyrights, optional text, and variable text is what led us in that direction. @myndzi did quite a bit of analysis on the approach.

**Comment by pombredanne** [comment]
ok, so I guess I can close this then. Thank you for chiming in.

**Comment by richardfontana** [comment]
This issue was closed nearly 7 years ago. Since that time, the trajectory of XML towards being a legacy data format has proceeded. I think @pombredanne's points in the issue continue to be valid. Recently I've been looking closely at the XML files in this repository and I'm troubled at how convoluted, inconsistent and haphazard the use of the XML seems to be in practice.

To @goneall's point that the XML is an "internal" format, I don't feel that is right, as someone who is external to the legal team. If people external to the SPDX legal team want to contribute to this project, they preferably need to create XML files. I think this (along with what seems to be a lack of significant documentation on the details of the XML format other than the fairly complex schema) is a large reason for the inconsistent usage of XML across the files in license-list-XML. If I want to write new tools to work with SPDX license data, it is most sensible to work with the XML files (since the XML is what's "official" and the other generated formats in general seem prone to having various subtle conversion errors). 

Finally, though this is maybe a separate issue, one of the reasons I'm seeing for the excessive complexity of the SPDX XML files seems to be efforts to make the XML generate sufficiently "pretty" HTML for purposes of the licenses displayed on https://spdx.org/licenses. Thus, for example, some XML files throw in <br/> tags in the license text even though this generally has nothing to do with the informational content of the license data. 

So I recommend that this issue be reopened.

**Comment by richardfontana** [comment]
One further comment about what @bradleeedmondson said: "@myndzi's response was that it was, at least if we wanted to be able to present the license text in a way that remotely resembles the original license text." I contend that this should not be a goal. The XML files are conceptually different from the original license text; in many cases "original license text" is not meaningful (see, e.g., `BSD-3-Clause` for a relatively extreme example). It seems to me that SPDX has adopted conflicting goals: (a) have a format for representing license *templates* against which real-world license-texts can be matched, and (b) have license data that can be displayed in ways that seem aesthetically pleasing. This seems to come up quite a bit in various ways in issues pertaining to new license submissions; for example, concerns are raised about designating text as `<optional>` because of how this will then show up in the display of text on https://spdx.org/licenses/ (which has its own significant problems, see e.g. https://github.com/spdx/LicenseListPublisher/issues/135).

**Comment by richardfontana** [comment]
I deleted some comments I made because I think it would be more productive for me to open a new issue than rant here. :)
